lib/Dialect/Triton/IR/Ops.cpp

```diff
@@ -719,29 +719,44 @@ LogicalResult ExpandDimsOp::canonicalize(ExpandDimsOp op,
                                                  splat.getOperand());
     return mlir::success();
   }
-  // expand_dims(broadcast) -> broadcast(expand_dims)
+  // expand_dims(broadcast(x)) -> broadcast(expand_dims(x))
   //
-  // On it's own this doesn't do much, but consider
+  // On its own this doesn't do much, but consider
   //    broadcast(expand_dims(broadcast))
   // -> broadcast(broadcast(expand_dims))
   // -> broadcast(expand_dims)
+  //
+  // Note we can't do this if x is a scalar, because expandDims requires a
+  // tensor input.  (We could change it to broadcast(expand_dims(splat(x))), but
+  // it's unclear that's better.)  broadcast currently accepts scalar inputs,
+  // but it probably shouldn't (just use splat instead).
```

Comment:
It's pretty weird for `tt.broadcast` to accept scalars given that it can't increase the rank of tensors.

Reply:
I agree!  (Thus the TODO in TritonOps.td.)

I am pretty busy rn -- is there anyone on your side who wants to take this, maybe as a starter project?

Reply:
cc @alexsamardzic @lezcano

Reply:
@alexsamardzic do you want to have a go at it?

Reply:
I'll take a look into this, thanks!